akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ShardingMessageExtractor.scala

```diff
@@ -91,17 +85,16 @@ final class HashCodeMessageExtractor[A](
  *
  * This is recommended since it does not force details about sharding into the entity protocol
  *
- * @tparam A The type of message accepted by the entity actor
+ * @tparam M The type of message accepted by the entity actor
  */
-abstract class HashCodeNoEnvelopeMessageExtractor[A](
-  val maxNumberOfShards:           Int,
-  override val handOffStopMessage: A)
-  extends ShardingMessageExtractor[A, A] {
+abstract class HashCodeNoEnvelopeMessageExtractor[M](
+  val numberOfShards: Int)
+  extends ShardingMessageExtractor[M, M] {
 
-  override def shardId(entityId: String): String = (math.abs(entityId.hashCode) % maxNumberOfShards).toString
-  override final def unwrapMessage(message: A): A = message
+  override def shardId(entityId: String): String = (math.abs(entityId.hashCode) % numberOfShards).toString
```

Comment:
How was it with the potentially negative output of math.abs, does it need a fix here?

Reply:
right, fixing in separate PR that can be merged more quickly https://github.com/akka/akka/pull/25542